pkg/ceph/osd/daemon.go

```diff
@@ -70,8 +68,8 @@ func Run(context *clusterd.Context, agent *OsdAgent) error {
 	}
 
 	// initialize the data directories, with the default dir if no devices were specified
-	devicesSpecified := len(devices.Entries) > 0
-	dirs, err := getDataDirs(context, agent.directories, devicesSpecified)
+	devicesSpecified := len(agent.devices) > 0
```

Comment:
Sure.  `devices.Entries` is the set of devices that passed through the `getAvailableDevices` check found here: https://github.com/rook/rook/blob/6307c2b56d9ed4ce61908d546c76856d2cbb72bb/pkg/ceph/osd/daemon.go#L109
These entries are only the devices that match the users request and are also open/available (no file system, no child partitions, etc.)

`agent.devices` is the set of devices (either a list or a filter) that the user specified.  That gets set on the agent from the `rook osd` entry point here: https://github.com/rook/rook/blob/6307c2b56d9ed4ce61908d546c76856d2cbb72bb/cmd/rook/osd.go#L88

What we are trying to capture here is if the user specified any devices at all, regardless of whether or not they are valid/available.  In that vein, using the `agent.devices`, the ones the user asked for originally, is the right thing to do.

Reply:
`devicesSpecified` is only trying to capture if the user specified any devices or not.  If they did not specify any devices then they don't want to use any and we won't use any.  The check for only `len(agent.devices) > 0` is sufficient because that exactly captures what we want: if the user specified any devices.